transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java

```diff
@@ -396,6 +397,11 @@ public BackendResponse send(BackendRequest request) {
     }
   }
 
+  @Override
+  public boolean shouldUploadClientHealthMetrics(TransportContext transportContext) {
+    return transportContext.getExtras() != null;
+  }
```

Comment:
>  CCT backend supports 2 different endpoints
>  We route to CCT backend if extras are null and to LegacyFlg otherwise.
>  This (anti-) pattern should not be required for other backends

[Reference](https://github.com/firebase/firebase-android-sdk/blob/2bb33648c3c3fae8fea3551226d098a133b9a82f/transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java#L351)


CCT and FLG are unified into one `TransportBackend`, it sends to FLG if the transportContext's extras is not null.

Reply:
Therefore, in order to only record events sent to Flg, `CctTransportBackend` overrides `shouldUploadClientHealthMetrics`, and
- return true if it sends request to Flg(transportContext's extras is not null)
- return false if it sends request to Cct(transportContext's extras is null)